docs/messaging.rst

```diff
@@ -256,8 +257,37 @@ All messages sent to or received by any IPython process should have this
 extended structure.
 
 
-Messages on the shell ROUTER/DEALER sockets
-===========================================
+Messages on the shell (ROUTER/DEALER) channel
+=============================================
+
+.. _request_reply:
+
+Request-Reply
+-------------
+
+In general, the ROUTER/DEALER sockets follow a request-reply pattern.
```

Comment:
Suggest adding a colon at the end of this line and then bulleting the individual process steps.